spec/packet_logs/packet_log_reader_spec.rb

```diff
@@ -278,6 +278,70 @@ module Cosmos
         end
       end
 
+      it "increments the command received count" do
+        plw = PacketLogWriter.new(:CMD,'cnt',true,nil,10000000,nil,false)
+        plw.write(System.commands.packet("INST","COLLECT").clone)
+        plw.write(System.commands.packet("INST","ABORT").clone)
+        plw.write(System.commands.packet("INST","ABORT").clone)
+        plw.write(System.commands.packet("INST","COLLECT").clone)
+        plw.write(System.commands.packet("COSMOS","STOPLOGGING").clone)
+        plw.write(System.commands.packet("INST","ABORT").clone)
+        plw.stop
+
+        cnt = {}
+        @plr.each(Dir[File.join(@log_path,"*cntcmd.bin")][0]) do |packet|
+          cnt["#{packet.target_name}_#{packet.packet_name}"] ||= 0
+          cnt["#{packet.target_name}_#{packet.packet_name}"] += 1
+          expect(packet.received_count).to eql cnt["#{packet.target_name}_#{packet.packet_name}"]
+        end
+
+        # Resetting a packet should reset only that packet's received_count
+        packet = System.commands.packet("INST","COLLECT").clone
+        packet.reset
+        cnt["INST_COLLET"] = 0
```

Comment:
Is it because you misspelled COLLECT here?   Also you are shadowing the packet variable which is not ideal/may not do what you expect